nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -356,6 +360,107 @@ public void testListTables() {
   }
 
+  @Test
+  public void testRegisterTableWithGivenBranch() {
+    List<String> metadataVersionFiles = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles.size());
+    catalog.dropTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"), false);
+    catalog.registerTable(
+            TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"),
+            "file:" + metadataVersionFiles.get(0));
+    Table newTable = catalog.loadTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"));
+    Assertions.assertThat(newTable).isNotNull();
+    TableOperations ops = ((HasTableOperations) newTable).operations();
+    String metadataLocation = ((NessieTableOperations) ops).currentMetadataLocation();
+    Assertions.assertThat("file:" + metadataVersionFiles.get(0)).isEqualTo(metadataLocation);
+    Assertions.assertThat(catalog.tableExists(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"))).isTrue();
+  }
+
+  @Test
+  public void testRegisterTableBranchNotPresent() {
+    List<String> metadataVersionFiles = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles.size());
+
+    Assertions.assertThatThrownBy(
+                    () -> catalog.registerTable(
+                            TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "`@default"),
+                            "file:" + metadataVersionFiles.get(0)))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Nessie ref 'default' does not exist");
+  }
+
+  @Test
+  public void testRegisterTableWithDefaultBranch() {
+    List<String> metadataVersionFiles = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles.size());
+    catalog.dropTable(TABLE_IDENTIFIER);
+    catalog.registerTable(TABLE_IDENTIFIER, "file:" + metadataVersionFiles.get(0));
+
+    Table newTable = catalog.loadTable(TABLE_IDENTIFIER);
+    Assertions.assertThat(newTable).isNotNull();
+    TableOperations ops = ((HasTableOperations) newTable).operations();
+    String metadataLocation = ((NessieTableOperations) ops).currentMetadataLocation();
+    Assertions.assertThat("file:" + metadataVersionFiles.get(0)).isEqualTo(metadataLocation);
+    Assertions.assertThat(catalog.tableExists(TABLE_IDENTIFIER)).isTrue();
+  }
+
+  @Test
+  public void testRegisterTableTableAlreadyExists() {
+    List<String> metadataVersionFiles = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles.size());
+
+    Assertions.assertThatThrownBy(() -> catalog.registerTable(TABLE_IDENTIFIER, "file:" + metadataVersionFiles.get(0)))
+            .isInstanceOf(AlreadyExistsException.class)
+            .hasMessage(
+                    "Table already exists: db.tbl");
+  }
+
+  @Test
+  public void testRegisterTableTagName() throws NessieConflictException, NessieNotFoundException {
```

Comment:
you can combine all 3 negative scenario test case (that is checking Assertions.assertThatThrownBy) into one.